src/markov/ddp.jl

```diff
@@ -501,6 +505,40 @@ function solve(ddp::DiscreteDP{T}, v_init::AbstractVector{T},
     ddpr
 end
 
+"""
+Solve a finite horizon discrete dynamic program by backward induction with
+stationary reward and transition probability functions and discount factor.
+
+##### Parameters
+
+- `ddp::DiscreteDP{T}` : Object that contains the Model Parameters
+- `J::Integer`: Number of decision periods
+- `v_term::AbstractVector{T}=zeros(num_states(ddp))`: Terminal value function,
+  of length equal to n (the number of states)  
+
+##### Returns
+
+- `vs::Array{Float64,2}`: Matrix that contains the optimal value function 
+  at each period
+- `sigmas::Array{Int,2}`: Matrix that contains the optimal policy function
+  at each period
+"""
+function backward_induction(ddp::DiscreteDP{T}, J::Integer,
+                            v_term::AbstractVector{T}=
+                            zeros(num_states(ddp))) where {T}
+    n = num_states(ddp)
+    vs = zeros(J+1,n)
```

Comment:
If `T` is `Rational`, the eltype of `vs` should be `Rational` too.

```suggestion
    S = typeof(zero(T)/one(T))
    vs = Matrix{S}(undef, J+1, n)
```